tests/test.ts

```diff
@@ -176,7 +176,8 @@ export function test(test: TestDescriptor) {
           try {
             await test.execute();
           } catch (e) {
-            logError(e)
+            logError(e);
+            fail("Uncaught exception in test\n" + e.message);
```

Comment:
I believe this breaks some tests because we have some mechanism to test the error message 
https://github.com/quarto-dev/quarto-cli/blob/ec4a95f0787bfc0f9ce1e925930fc7b3e4d35224/tests/verify.ts#L95-L109

Now, any test using `printsMessage()` as a verify function will fail.

Reply:
Hm. That's fair and I'll revert it, but this seems like the kind of thing we should fix. Because if we have a test that fails with an uncaught exception, it currently looks like the test passes.

Reply:
What is an uncaught exception here ? 

I though we had `noErrors` and `noErrorsOrWarnings` to use in tests when we want to fail test on exceptions 
https://github.com/quarto-dev/quarto-cli/blob/ec4a95f0787bfc0f9ce1e925930fc7b3e4d35224/tests/verify.ts#L45-L93

Our smoke-test workflow defaults on `noErrorsOrWarnings` and allow to specify `noErrors` in `_quarto.tests`. 
https://github.com/quarto-dev/quarto-cli/blob/ec4a95f0787bfc0f9ce1e925930fc7b3e4d35224/tests/smoke/smoke-all.test.ts#L104-L115

https://github.com/quarto-dev/quarto-cli/blob/ec4a95f0787bfc0f9ce1e925930fc7b3e4d35224/tests/smoke/smoke-all.test.ts#L145-L147

In non smoke-test, we default to `noErrorsOrWarnings` in `testProjectRender()`, but in other function either `noErrorsOrWarnings` or `noErrors` usually needs to be added among verify functions. 

Hopefully, this is relevant, and I understood correctly your feedback, and it helps.

Reply:
I had a bug in this test https://github.com/quarto-dev/quarto-cli/blob/04524462d5653b57969db2daa49a0b4dfc13bc19/tests/smoke/jupyter/cache.test.ts where I used the wrong path, and so `await quarto` threw an exception. (Locally for me at least) the test passed, because we don't fail tests with uncaught exceptions.

Reply:
I see. My understanding is that if you use `verify: [noErrorsOrWarnings],` instead of `verify: [],`, the error would have been caught. Maybe we should have another test wrapper that does default to this.

Reply:
I agree. At least `testQuartoCmd()` should have this as default it seems. `test()` is more generic, but I agree it would be better to have it by default too.  

We could try change our functions to add a default and see if the tests suites still pass ? 🤷‍♂️